rasa/utils/tensorflow/models.py

```diff
@@ -424,6 +482,35 @@ def _log_metrics_for_tensorboard(
                     if metric.name in self.metrics_to_log:
                         tf.summary.scalar(metric.name, metric.result(), step=step)
 
+    def _does_model_improve(self, current_results: Dict[Text, Text]) -> bool:
+        # Initialize best_metrics_so_far with the first results
+        if not self.best_metrics_so_far:
+            keys = filter(
+                lambda k: True if (k.endswith("_acc") or k.endswith("_f1")) else False,
+                current_results.keys(),
+            )
+            for key in keys:
+                self.best_metrics_so_far[key] = float(current_results[key])
+            return True
+
+        all_improved = all(
+            [
+                float(current_results[key]) > self.best_metrics_so_far[key]
+                for key in self.best_metrics_so_far.keys()
+            ]
+        )
+        if all_improved:
+            for key in self.best_metrics_so_far.keys():
+                self.best_metrics_so_far[key] = float(current_results[key])
+        return all_improved
+
+    def _save_model_checkpoint(self, current_results: Dict[Text, Text], epoch: int
+    ) -> None:
+        if self.checkpoint_model and self._does_model_improve(current_results):
+            logger.debug(f"Creating model checkpoint at epoch={epoch}...")
+            self.best_model_epoch = epoch
```

Comment:
As epoch starts with 0, we might want to do 
```
            self.best_model_epoch = epoch + 1
```
so that we don't log something like `Creating model checkpoint at epoch=0...`